uproot/rootio.py

```diff
@@ -300,6 +304,9 @@ def items(self, recursive=False, filtername=nofilter, filterclass=nofilter):
     def classes(self, recursive=False, filtername=nofilter, filterclass=nofilter):
         return list(self.iterclasses(recursive=recursive, filtername=filtername, filterclass=filterclass))
 
+    def classnames(self):
+        return list(key._fClassName.decode('ascii') for key in self._keys)
```

Comment:
This one should probably be a list of 2-tuples and take `recursive`, `filtername`, `filterclass` methods, like `classes`. That would make it easier to match up the output with the keys.

```python
    def iterclassnames(self, recursive=False, filtername=nofilter, filterclass=nofilter):
        for key in self._keys:
            if filtername(key._fName) and filterclass(cls):
                yield self._withcycle(key), key._fClassName
            if recursive and (key._fClassName == b"TDirectory" or key._fClassName == b"TDirectoryFile"):
                for name, classname in key.get().iterclassnames(recursive, filtername, filterclass):
                    yield "{0}/{1}".format(self._withoutcycle(key).decode("ascii"), name.decode("ascii")).encode("ascii"), classname

    def classnames(self, recursive=False, filtername=nofilter, filterclass=nofilter):
        return list(self.iterclassnames(recursive=recursive, filtername=filtername, filterclass=filterclass))

    def allclassnames(self, filtername=nofilter, filterclass=nofilter):
        return self.classnames(recursive=True, filtername=filtername, filterclass=filterclass)
```

Reply:
Maybe "classes" was the wrong name for this (because that implies just the classes, not also the key names), but we need the correlation because it's safer than trusting that we always get these keys in the same order. It would be nicer if it came back as a dict, but for many versions of Python, dicts don't preserve order, and I'd like to at least _try_ to preserve order.

Maybe that's not a great argument. I'm open to changing that interface in uproot 4, but for this PR, I'd like the interfaces to be consistent.

Reply:
Okay, I updated this. But is it not guaranteed that `.keys()` will return the same order as `.classes()`? I would have expected to do `for key, class in zip(f.keys(), f.classes()` if I wanted to match keys to classes (or class names).